I'll go ahead and change it to the string version then. That's odd though
because somehow that's a different numpy version

On Tue, Mar 12, 2024, 16:21 Kiyoto Yabe ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In python/pfs/drp/qa/utils/helpers.py
> <https://github.com/Subaru-PFS/drp_qa/pull/9#discussion_r1522391519>:
>
> > @@ -239,10 +240,10 @@ def getTargetType(arc_data, pfsConfig) -> pd.DataFrame:
>
>
>  def getFitStats(
> -    arcLines: ArcLineSet,
> -    detectorMap: DetectorMap,
> -    selection: np.ndarray[bool],
> -    numParams: int = 0):
> +        arcLines: ArcLineSet,
> +        detectorMap: DetectorMap,
> +        selection: NDArray[Bool],
>
> I actually got the problem like this:
>
> (lsst-scipipe-3.0.0) detectorMapQa.py --help
> Traceback (most recent call last):
>   File "/work/kiyoyabe/erun/run15/calibs/src/test/drp_qa/bin/detectorMapQa.py", line 2, in <module>
>     from pfs.drp.qa.detectorMapQa import DetectorMapQaTask
>   File "/work/kiyoyabe/erun/run15/calibs/src/test/drp_qa/python/pfs/drp/qa/detectorMapQa.py", line 27, in <module>
>     from pfs.drp.qa.utils import helpers
>   File "/work/kiyoyabe/erun/run15/calibs/src/test/drp_qa/python/pfs/drp/qa/utils/helpers.py", line 9, in <module>
>     from numpy.typing import NDArray, Bool
> ImportError: cannot import name 'NDArray' from 'numpy.typing' (/work/stack_INFRA-312/conda/miniconda3-py38_4.9.2/envs/lsst-scipipe-3.0.0/lib/python3.8/site-packages/numpy/typing/__init__.py)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Subaru-PFS/drp_qa/pull/9#discussion_r1522391519>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAARIUHMME3ZK73NDCXOOW3YX6ZZ3AVCNFSM6AAAAABERR667SVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZTGA2DQMBZHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

--- a/python/pfs/drp/qa/utils/helpers.py
+++ b/python/pfs/drp/qa/utils/helpers.py
@@ -239,10 +239,10 @@ def getTargetType(arc_data, pfsConfig) -> pd.DataFrame:
 
 
 def getFitStats(
-    arcLines: ArcLineSet,
-    detectorMap: DetectorMap,
-    selection: np.ndarray[bool],
-    numParams: int = 0):
+        arcLines: ArcLineSet,
+        detectorMap: DetectorMap,
+        selection: 'np.ndarray[bool]',
+        numParams: int = 0):
     """Gets output from calculateFitStatistics
 
     Parameters

